Riot/Modules/Room/TimelineDecorations/ReadReceipts/MXKReceiptSendersContainer.h

```diff
@@ -93,7 +98,7 @@ typedef NS_ENUM(NSInteger, ReadReceiptsAlignment)
  @param placeHolders list of placeholders, one by room member. Used when url is nil, or during avatar download.
  @param alignment (see ReadReceiptsAlignment).
  */
-- (void)refreshReceiptSenders:(NSArray<MXRoomMember*>*)roomMembers withPlaceHolders:(NSArray<UIImage*>*)placeHolders andAlignment:(ReadReceiptsAlignment)alignment;
+- (void)refreshReceiptSenders:(NSArray<MXRoomMember*>*)roomMembers nonRoomMembersReceiptSenders: (NSArray<MXUser*>*)nonRoomMembers withPlaceHolders:(NSArray<UIImage*>*)placeHolders andAlignment:(ReadReceiptsAlignment)alignment;
```

Comment:
Passing 2 different "member" arrays but keeping one single placeholders array for both of them is hard to manage (see `combinedIndex`) and quite error prone. I would suggest we switch it to
```
- (void)refreshWithMemberAvatarURLStrings:(NSArray<NSString *>*)avatar
                             placeholders:(NSArray<UIImage*>*)placeholders
                                alignment:(ReadReceiptsAlignment)alignment;
```

This way the `MXKReceiptSendersContainer` becomes simpler from both the types its aware of but also from how it manages avatars and placeholders. It becomes the caller's responsibility to gather the right information for it to use.

Reply:
This of course means changing existing properties to `NSArray <NSString *>` and refactoring things.